ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -356,14 +367,23 @@ private static void ensureKeySchemasMatch(
       return;
     }
 
+    final SchemaRegistryClient schemaRegistryClient = serviceContext.getSchemaRegistryClient();
+
+    final FormatInfo formatInfo = addSerializerMissingFormatFields(
```

Comment:
Prior to this PR, insets only worked with SR schema using a default name. If a non-default name was used, then INSERT fail.

The problem is that  when calling the Connect API to serialize the row, the API looks at the row schema and checks with SR that the schema exists. If the row schema has a different name than SR, then the API fails. If the row schema does not have a name, then the API uses a default one. The fix is to add the full name to the schema before calling the API.

I couldn't make the API to use the SR schema to serialize the row. The API validates the SR schema matches the row KSQL intents to serialize.

Reply:
IIUC, you're saying that the Connect APIs validate that the schema object we pass matches what's in SR and fails if it doesn't. If that's the case, then we have a problem beyond just figuring out the base schema name, right? The logic in AvroSchemas right now has logic for creating additional schema names (for maps and structs) in a predetermined way from the base schema name provided. If the user has a schema that doesn't fit this pattern, then even if we fetch the base schema name from SR, we won't have a fully matching schema afterwards and the insertion will still fail.

I think the only foolproof way to get around this is to fetch the entire schema from SR and use that for serialization (possibly converting the fetched schema into a Connect schema first, if that's required by the Connect APIs). If we don't want to fetch on every insertion (as @lihaosky pointed out in another comment), then we'll have to store the schema somehow. Either way, this seems like it might be a larger conversation with product implications -- we could take the approach in this PR and solve the issue for (hopefully) most use cases, but we won't have a complete solution unless we do something more drastic.

I wonder why the Connect APIs we use are so rigid. Does it make sense for us to consider relaxing requirements from their API (with a contribution ourselves) as another option around these challenges?